src/types.deno.ts

```diff
@@ -114,19 +114,19 @@ export class InputFile {
                     "Reading files by path requires a Deno environment",
                 );
             }
-            const file = await Deno.open(data);
-            return iterateReader(file);
+            using file = await Deno.open(data);
```

Comment:
I have one last question about this. My understanding is that `using` causes the file to be closed as soon as the function returns (end of scope), not when the file contents are fully read. How am I wrong about this?

Reply:
Your understanding is correct. Using `Deno.open` opens a file and returns a file descriptor, which is automatically closed when the scope ends, not necessarily when the file contents are fully read.

Reply:
But returned value is `file.readable` that is a Objejct (like **pointer**), gc will not immediately release these resource.

Reply:
Yeah, you're right. But the design of JavaScript sucks ass (The ownership is a complete mess. Objects are passed around carelessly, and no one knows where to release them.). But it does work 🤡:)

Reply:
Rustacean spotted

Reply:
* When I update to the latest version of Deno, writing it like this does cause issues. 
* I will update the PR. :)